### PR TITLE
85 create responsive page layout component

### PR DESCRIPTION
**Pull request by SarahEddeb**
Home page desktop view:
image
Home page mobile view:
image

Lesson page desktop view:
image
Lesson page mobile view:
image

Sidebar desktop view:
image
Sidebar mobile view:
image

### CODE REVIEW

**Review by SarahEddeb — COMMENTED** [review]
Changes include:
- Minor style adjustments in Button, ChapterButtonCard, ChapterListHeader, IconButton, Label, LessonButtonCard, Sidebar, and Typography components
- Adjust styling and positioning of child components in VideoPlayerSection for cleaner design
- Change design of VideoHeader text to have Arabic and English titles next to each other separated by a straight line
- Allow Navbar to collapse or expand based on users scrolling behaviour 
- Create ResponsiveContainer component (and storybook file) that adjusts margins and padding based on screen size
- Breaking down index.tsx into HomePage.tsx and LessonPage.tsx
- Update mock data file with Quran Journey notes pdf
- Create SidebarDropdown component (and storybook file) that handles the expansion and collapse of the Sidebar component. 

Todo:
- Create fetch calls in HomePage and LessonPage

File: src/components/NavBar/NavBar.tsx [hunk 1]
```diff
@@ -1,37 +1,85 @@
...
+import React, { useState, useEffect } from 'react';
 import styled from 'styled-components';
 import { breakpoints } from '../../styles/breakpoints';
 import { colours } from '../../styles/colours';
 import { Typography } from '../Typography/Typography';
 
+/**
+ * Interface for the NavBarProps.
+ * @property {boolean} isVisible - Indicates whether the navigation bar is visible or not.
+ */
```
Comment (aeldamaty) [inline src/components/NavBar/NavBar.tsx:10]: This code here is pretty clear. There's probably no need for these comments here, unless these are used for auto generated docs?

File: src/components/ResponsivePage/ResponsiveContainer.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import type { Meta, StoryObj } from '@storybook/react';
```
Comment (aeldamaty) [inline src/components/ResponsivePage/ResponsiveContainer.stories.tsx:1]: What's the point of storybook for us again? I have heard of it, but didn't dig too deep.

File: src/pages/HomePage/HomePage.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import { ResponsiveContainer } from '@/components/ResponsivePage/ResponsiveContainer';
+import { Sidebar, SidebarProps } from '@/components/Sidebar/Sidebar';
+import { NavBar } from '@/components/NavBar/NavBar';
+
+// TODO: Remove once fetch calls in place
```
Comment (aeldamaty) [inline src/pages/HomePage/HomePage.tsx:6]: I thought we were saying we are now using the actual fetch calls from the API no? Or am I confusing that with something else?

**Review by aeldamaty — APPROVED** [review]
Overall looks good. Couple of high level things:
1. We can likely cut down on comments overall. Commenting should be for sections of code that are more difficult to understand. Generally, you should think of it like this: Is this code easy to understand? Yes? Cool no comments. No? Okay can I make it easier to understand and follow? If so, do that, if not, then add comments.
2. From what I remember from the call, I thought we were calling the API no? Or is that a separate PR?
3. Let's make sure we squash all these commits into 1 commit. 
Giving an approval, none of these comments are blocking comments

File: src/components/ResponsivePage/ResponsiveContainer.stories.tsx (hunk 2 above)
Comment (SarahEddeb) [reply src/components/ResponsivePage/ResponsiveContainer.stories.tsx:1]: Storybook is used to keep track of the components outside of the development environment. Its easy to look through and see what has been made for the frontend.

File: src/pages/HomePage/HomePage.tsx (hunk 3 above)
Comment (SarahEddeb) [reply src/pages/HomePage/HomePage.tsx:6]: Yes we are. That will be in a separate PR as this one is mainly for the responsive container component and other styling changes.

File: src/components/NavBar/NavBar.tsx (hunk 1 above)
Comment (SarahEddeb) [reply src/components/NavBar/NavBar.tsx:10]: Some comments are used for docs. In general, I commented more than usual as we were considering bringing on interns to the frontend and most members on the team were more familiar with backend code than the frontend.

**Pull request merged**